domains/certificates/Query.API/API.IntegrationTests/Testcontainers/ProjectOriginStack.cs

```diff
@@ -28,16 +31,19 @@ public ProjectOriginStack()
         {
             var connectionString = $"Host={postgresContainer.IpAddress};Port=5432;Database=postgres;Username=postgres;Password=postgres";
 
-            // The host port is fixed due to the fact that it used in the value for "ServiceOptions__EndpointAddress"
-            // There is a chance for port collision with the host ports assigned by Testcontainers
+            // Get an available port from system and use that as the host port
+            var udp = new UdpClient(0, AddressFamily.InterNetwork);
+            var hostPort = ((IPEndPoint)udp.Client.LocalEndPoint!).Port;
+
             return new ContainerBuilder()
-                .WithImage("ghcr.io/project-origin/wallet-server:0.1.0-rc.7")
-                .WithPortBinding(7890, GrpcPort)
+                .WithImage($"ghcr.io/project-origin/wallet-server:{WalletVersion.Get()}")
+                .WithPortBinding(hostPort, GrpcPort)
                 .WithCommand("--serve", "--migrate")
                 .WithEnvironment("ConnectionStrings__Database", connectionString)
-                .WithEnvironment("ServiceOptions__EndpointAddress", "http://localhost:7890/")
+                .WithEnvironment("ServiceOptions__EndpointAddress", $"http://localhost:{hostPort}/")
```

Comment:
I am curious as to if you managed to find a "smarter way", to assign host ports to the wallet service, which can be used for other domains, that use the wallet service in a test container?